### PR TITLE
bpo-45209: fix `UserWarning: resource_tracker` in test_multiprocessing

### PR DESCRIPTION
**Pull request by sobolevn**
https://bugs.python.org/issue45209

Снимок экрана 2021-09-15 в 23 40 49

### CODE REVIEW

**Comment by vstinner** [comment]
cc @pitrou

File: Misc/NEWS.d/next/Tests/2021-09-15-23-32-39.bpo-45209.55ntL5.rst [hunk 1]
```diff
@@ -0,0 +1,2 @@
+Fixes ``UserWarning: resource_tracker`` warning in
```
Comment (pitrou) [inline Misc/NEWS.d/next/Tests/2021-09-15-23-32-39.bpo-45209.55ntL5.rst]: "Fix"

Comment (sobolevn) [reply Misc/NEWS.d/next/Tests/2021-09-15-23-32-39.bpo-45209.55ntL5.rst]: 👍

**Comment by sobolevn** [comment]
All PRs right now have this problem: https://dev.azure.com/Python/cpython/_build/results?buildId=87889&view=logs&j=91c152bd-7320-5194-b252-1404e56e2478&t=c7e99cd8-4756-5292-d34b-246ff5fc615f

Another examples:
- https://github.com/python/cpython/pull/28387
- https://github.com/python/cpython/pull/28382

**Comment by vstinner** [comment]
Windows build error on Azure Pipelines is tracked by: https://bugs.python.org/issue45220

**Comment by sobolevn** [comment]
Rebased, CI now passes.

File: Misc/NEWS.d/next/Tests/2021-09-15-23-32-39.bpo-45209.55ntL5.rst [hunk 2]
```diff
@@ -0,0 +1,2 @@
+Fix ``UserWarning: resource_tracker`` warning in
```
Comment (vstinner) [inline Misc/NEWS.d/next/Tests/2021-09-15-23-32-39.bpo-45209.55ntL5.rst:1]: You should mention that the fix is for multiprocessing tests.

**Review by vstinner — APPROVED** [review]
LGTM. @pitrou: Would you mind to double check this multiprocessing test fix?

I'm kind of unhappy that you have to prepend "/". SharedMemory._prepend_leading_slash does a magic dance (add/remove "/" prefix) with SharedMemory(name) constructor parameter and in SharedMemory.name property. But I don't think that it's worth it to add something just for this very specific test. We can always revisit the code later if we start to have issue on some corner cases.

**Comment by sobolevn** [comment]
> I'm kind of unhappy that you have to prepend "/". SharedMemory._prepend_leading_slash does a magic dance (add/remove "/" prefix) with SharedMemory(name) constructor parameter and in SharedMemory.name property

Yes, me too. I've tried several other things like `smm.name` (was not producing the correct result by raising `KeyError`). After several attempts this was the simplest solution. And since we only run this part on posix, it should work correctly.

Thanks, @vstinner and @pitrou

**Comment by ambv** [comment]
Since the slash hack is tests-only, let's have it. We can fix the fix later if we find a way to do so.

**Pull request merged**